test/e2e/node/runtimeclass.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/node/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclasstest "k8s.io/kubernetes/pkg/kubelet/runtimeclass/testing"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	"k8s.io/kubernetes/test/e2e/scheduling"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	utilpointer "k8s.io/utils/pointer"
+
+	"github.com/onsi/ginkgo"
+)
+
+var _ = ginkgo.Describe("[sig-node] RuntimeClass", func() {
+	f := framework.NewDefaultFramework("runtimeclass")
+
+	ginkgo.It("should reject a Pod requesting a RuntimeClass with conflict node selector", func() {
+		handler := framework.GetRuntimeClassHandler("test-handler")
+		scheduling := &v1beta1.Scheduling{
+			NodeSelector: map[string]string{
+				"foo": "conflict",
+			},
+		}
+
+		runtimeClass := newRuntimeClass(f.Namespace.Name, "conflict-runtimeclass", handler)
+		runtimeClass.Scheduling = scheduling
+		rc, err := f.ClientSet.NodeV1beta1().RuntimeClasses().Create(runtimeClass)
+		framework.ExpectNoError(err, "failed to create RuntimeClass resource")
+
+		pod := newRuntimeClassPod(rc.GetName())
+		pod.Spec.NodeSelector = map[string]string{
+			"foo": "bar",
+		}
+		_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
+		framework.ExpectError(err)
```

Comment:
Assert that it's a forbidden error. You can copy the non-node portion of the `expectRejection` function from https://github.com/kubernetes/kubernetes/pull/82094 to this test.